### PR TITLE
Support cellular status.

### PR DESCRIPTION
**Pull request by maxh1981**

### CODE REVIEW

File: cellular_utility/management.py [hunk 1]
```diff
@@ -20,20 +20,60 @@
...
+            signal=None,
+            operator=None,
+            lac=None,
+            cell_id=None,
+            icc_id=None,
+            imei=None,
+            pin_retry_remain=None):
+        self._sim_status = sim_status
+
+        if mode is None:
```
Comment (imZack) [inline cellular_utility/management.py]: how about this:
`mode = "" if mode is None else mode`

Comment (maxh1981) [inline cellular_utility/management.py]: Great, that's exactly what I need!

**Comment by maxh1981** [comment]
Fix a lot of stuff...

File: cellular_utility/cell_mgmt.py [hunk 2]
```diff
@@ -303,16 +372,21 @@ def m_info(self):
...
-            "WWAN_node": match.group(2),
-            "LAC": match.group(3),
-            "CellID": match.group(4),
-            "ICC-ID": match.group(5),
-            "IMEI": match.group(6)
-        }
+        qmi_port = match.group(7)
+        if qmi_port == "":
+            qmi_port = None
+
```
Comment (imZack) [inline cellular_utility/cell_mgmt.py]: qmi_port = "" if qmi_port == "" else None

File: cellular_utility/cell_mgmt.py [hunk 3]
```diff
@@ -332,6 +406,7 @@ def operator(self):
 
         match = self._operator_regex.match(output)
         if not match:
+            _logger.warning("unexpected output: " + output)
```
Comment (imZack) [inline cellular_utility/cell_mgmt.py]:
```py
_logger.warning("unexpected output: {}".format(output))
```

File: cellular_utility/cell_mgmt.py [hunk 4]
```diff
@@ -375,15 +447,41 @@ def sim_status(self):
...
+            return -1
+
+        _logger.debug("qmicli -d " + qmi_port + " --dms-uim-get-pin-status")
+        output = check_output(
+            ["qmicli", "-d", qmi_port, "--dms-uim-get-pin-status"],
+            shell=self._use_shell)
+
+        match = CellMgmt._pin_retry_remain_regex.match(output)
+        if not match:
+            _logger.warning("unexpected output: " + output)
```
Comment (imZack) [inline cellular_utility/cell_mgmt.py]:
```py
_logger.warning("unexpected output: {}".format(output))
```

File: cellular_utility/management.py [hunk 5]
```diff
@@ -331,24 +385,29 @@ def _power_cycle(self):
...
 
-        # wait another 10 seconds to ensure module readiness
-        sleep(10)
+        # wait another few seconds to ensure module readiness
+        sleep(30)
 
         sim_status = self._cell_mgmt.sim_status()
-        if sim_status == "nosim":
+        _logger.debug(
+            "sim_status = " + sim_status.name + ", self._pin = " + self._pin)
```
Comment (imZack) [inline cellular_utility/management.py]:
```py
 _logger.debug("sim_status = {}, self._pin = {}".format(sim_status.name, self._pin))
```

**Comment by imZack** [comment]
:sparkles:  Note: since this commit has been merged, cellular bundle only supports **MC7304** module.

**Pull request merged**